### PR TITLE
fix: Improves comments section UI

### LINKED ISSUE #1615: [Bug]: Replies are too far from original comment
### What happened?

CleanShot 2024-10-31 at 15 58 47@2x

We refactored the comment section recently and seems replies have been pushed too far away from original comment

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by satishkumarsajjan**
## Description

Changed comment component as discussed in the Issue #1615 and #1629 

## Related Issue

Closes #1615 and may be #1629 

## Motivation and Context

UI was out of sync

## Screenshots/Video (if applicable):
[Screencast from 2024-11-12 20-20-12.webm](https://github.com/user-attachments/assets/ff150c9f-b5ca-47ef-bc32-f765e530ca78)

### CODE REVIEW

**Comment by bautistaaa** [comment]
can you run `pnpm checks:fix`

going to get this merged first: https://github.com/typehero/typehero/pull/1640

then we can get yours in!

**Comment by bautistaaa** [comment]
> can you run `pnpm checks:fix`
> 
> going to get this merged first: #1640
> 
> then we can get yours in!

merged, need to resolve the conflicts

File: apps/web/src/app/challenge/_components/comments/comment.tsx [hunk 1]
```diff
@@ -280,25 +287,125 @@ function SingleComment({
...
+              </DropdownMenuContent>
+            </DropdownMenu>
+          </div>
         </div>
       </div>
 
       <div className="flex gap-3">
         <ExpandableContent content={comment.text} />
-        <div className="flex flex-col items-end gap-2">
+        {/* <div className="flex flex-col items-end gap-2">
```
Comment (aquibbaig) [inline apps/web/src/app/challenge/_components/comments/comment.tsx:408]: let's remove this comment (if not needed)

File: apps/web/src/app/challenge/_components/comments/comment.tsx [hunk 2]
```diff
@@ -490,22 +561,41 @@ function ExpandableContent({ content }: { content: string }) {
...
+            setExpanded(true);
+            setSeeCollapse(true);
+          }}
         >
           <div className="text-md text-label-1 dark:text-dark-label-1 flex w-full items-center justify-center hover:bg-transparent">
             Read more
           </div>
         </div>
       )}
+      {expanded && seeCollapse && (
```
Comment (aquibbaig) [inline apps/web/src/app/challenge/_components/comments/comment.tsx:583]: do we actually need a new state `collapse` here. I think `expanded` does  something similar.

If we need a new state, shouldn't this condition be `expanded || seeCollapse` to keep the previous behaviour intact ?

**Review by aquibbaig — CHANGES_REQUESTED** [review]

**Pull request closed without merging**